modules/nf-core/bcftools/norm/main.nf

```diff
@@ -12,19 +12,28 @@ process BCFTOOLS_NORM {
     path(fasta)
 
     output:
-    tuple val(meta), path("*.gz") , emit: vcf
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("*.vcf")      , emit: vcf     , optional: true
+    tuple val(meta), path("*.vcf.gz")   , emit: vcfgz   , optional: true
+    tuple val(meta), path("*.bcf")      , emit: bcf     , optional: true
+    tuple val(meta), path("*.bcf.gz")   , emit: bcfgz   , optional: true
```

Comment:
I think it would be better to output everything in one output channel here since it's always one of the four. So use `path("*.{vcf,vcf.gz,bcf,bcf.gz}")` here :)

Reply:
I agree.

Reply:
hm that is very inconsistent with how we treat the samtools modules for example. There it will also always be bam or cram, but we still output them in separate channels

Reply:
But yeah wouldn't it be more flexible if we didn't do that? I've seen some subwfs that contain bottlenecks because of this during the spring cleaning

Reply:
Thanks for your recommendations! I made a new update.

Reply:
Thanks can you wait with the merge until @FriederikeHanssen agrees too?

Reply:
Absolutely! Sorry I haven't seen the previous messages. In my opinion, I think @FriederikeHanssen suggestion is tidier but I do not have much experience and am not aware about the bottlenecks.

Reply:
Here is an example: https://github.com/nf-core/modules/blob/ffa9641ee18f88aff974257cb50ba3cf8f7d143c/subworkflows/nf-core/bam_dedup_stats_samtools_umitools/main.nf#L30-L32.
Here the join statement use remainder true because it can't know whether a `.bai` or `.crai` file has been created. The pipeline will wait here until all `SAMTOOLS_INDEX` processes are run before creating the channels with `remainder`. The pipeline won't be on hold here if `SAMTOOLS_INDEX` were to have one channel for `bai` and `crai` (e.g. `index`).

Reply:
ok, maybe you bring it up quickly in maintainers or modules, so we can add it as a guideline? I think this will impact a whole bunch of modules, and at that time we probably didn't consider subworkflows where you don't know the data type

Reply:
I'll post about it on Slack, thanks!

Reply:
You should know if you expect a bai or a csi, I'm assuming you have a params for that

Reply:
but I see your point, and it could be interesting to have just index as output to samtools to capture both possiblity of bai/csi.
It's true that they are different files, but they are created in the same logic for the same purpose

Reply:
Yeah but that's not the case in that subworkflow. And what about `.bai` and `.crai`? That's entirely dependent on the input file

Reply:
> but I see your point, and it could be interesting to have just index as output to samtools to capture both possiblity of bai/csi. It's true that they are different files, but they are created in the same logic for the same purpose

Yes! But we should definitely get a lot of eyes on this :) as Rike said, it will be a breaking change

Reply:
What would you prefer me to do with this module for now?